ynput/ayon-core
# Substance Painter: Support for exporting maps/layer stacks with specific channels

## Changelog Description
This PR is to add support for exporting outputs with specific channels defined in the settings and the users can choose to export layer with `use_selection` being enabled.
![image](https://github.com/ynput/ayon-core/assets/64118225/08677699-20e5-4528-b880-29a12f2b256c)
![image](https://github.com/ynput/ayon-core/assets/64118225/da704d37-cf6d-40e3-a8b1-1e7bd565ea91)
User can add/define their channel map in the ayon project setting, once they add their maps, it would show up in the multiselection options of export channel. 
You need to specify your channel type and the name of your channel type(of which name extension you set in your export presets)
![image](https://github.com/ynput/ayon-core/assets/64118225/a2142d63-1d4e-41c2-80c3-c20915967ca0)

## Additional info
- Update SubstancePainter Addon 0.1.2 to test this PR
- Export Layer Stack by selection is only compatible with substance painter 10.0.0
- Layer Stack Export only works when both export channel and use selections being enabled.
The template below showed the setting does not support for exporting specific channels
- Export specific channels are not supported for the templates from the export-preset-generator due to the limitation of Substance API (currently asking Substance Painter support in substance discord now to see if they would support in the next version)
```
"2D View",
"Document channels + Normal + AO (No Alpha)",  
"Document channels + Normal + AO (With Alpha)", 
"Sketchfab",  
"Substance 3D Stager",
 "USD PBR Metal Roughness", 
"glTF PBR Metal Roughness",  
"glTF PBR Metal Roughness + Displacement texture (experimental)" 

```

## Testing notes:
Without layer stack
1. Create Texture Set with your preferred template
2. Choose your export channel
3. Publish
4. Only your chosen export channel is being published

With selected layer stack
1. Create Texture Set with your preferred template with your selected layer stack with the `use_selection` being enabled.
2. Choose your export channel
3. Publish
4. Only your selected layer stack with your chosen export channel being published

General
1. Create Texture Set
2. Export Channel without any input
3. Publish
4. All channels should be published correctly
